partner/WHMCS/modules/servers/jetpack/jetpack.php

```diff
@@ -71,18 +71,32 @@ function jetpack_ConfigOptions()
  */
 function jetpack_CreateAccount(array $params)
 {
-
-    validate_required_fields($params);
-    $access_token = get_access_token($params);
-    $response = provision_jetpack_plan($access_token, $params);
-
-    if (isset($response->next_url)) {
-        save_next_url($response->next_url, $params);
+    try {
+        validate_required_fields($params);
+
+        $access_token = get_access_token($params);
+        $provisioning_url = "https://public-api.wordpress.com/rest/v1.3/jpphp/provision";
+        $request_data = array (
+            'plan' => strtolower($params['customfields']['Plan']),
+            'siteurl' => $params['customfields']['Site URL'],
+            'local_user' => $params['customfields']['Local User'],
+            'force_register' => true,
+        );
+
+        $response = make_api_request($provisioning_url, $access_token, $request_data);
+
+        if (isset($response->next_url)) {
+            save_next_url($response->next_url, $params);
+        }
+
+        if ($response->success == true) {
+            return 'success';
+        }
+    } catch(Exception $e) {
+        logModuleCall('jetpack', __FUNCTION__, $params, $e->getMessage(), $e->getMessage());
```

Comment:
Isn't there a better logging function than `logModuleCall`? From [the docs](https://developers.whmcs.com/provisioning-modules/module-logging/) it looks like the function's semantics are a bit different, not so much to log errors, but to log external API calls and requests.

I am curious, will the error be logged somewhere if we just return a non-“success” response in this function?